frontend/packages/kubevirt-plugin/src/components/vm-disks/guest-agent-file-systems.tsx

```diff
@@ -0,0 +1,130 @@
+import { NO_GUEST_AGENT_MESSAGE, VIRTUAL_MACHINE_IS_NOT_RUNNING } from '../../strings/vm/messages';
+
+import './guest-agent-file-systems.scss';
+
+const tableColumnClasses = [
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-sm-4'),
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-sm-4'),
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-sm-4'),
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-sm-4'),
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-sm-4'),
+];
+
+const FileSystemsTableHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'metadata.name',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'File System Type',
+      sortField: 'metadata.fileSystemType',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Mount Point',
+      sortField: 'metadata.mountPoint',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: 'Total Bytes',
+      sortField: 'metadata.totalBytes',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[3] },
+    },
+    {
+      title: 'Used Bytes',
+      sortField: 'metadata.usedBytes',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[4] },
+    },
+  ];
+};
+FileSystemsTableHeader.displayName = 'FileSystemsTableHeader';
+
+const FileSystemTableRow = ({ obj: fileSystem, index, key, style }) => {
+  return (
+    <TableRow id={fileSystem?.metadata?.uid} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>{fileSystem?.metadata?.name}</TableData>
+      <TableData className={tableColumnClasses[1]}>
+        {fileSystem?.metadata?.fileSystemType}
+      </TableData>
+      <TableData className={tableColumnClasses[2]}>{fileSystem?.metadata?.mountPoint}</TableData>
+      <TableData className={tableColumnClasses[3]}>{fileSystem?.metadata?.totalBytes}</TableData>
+      <TableData className={tableColumnClasses[4]}>{fileSystem?.metadata?.usedBytes}</TableData>
+    </TableRow>
+  );
+};
+
+export const FileSystemsList: React.FC<FileSystemsListProps> = ({ vmi, vmStatusBundle, delay }) => {
+  const [guestAgentInfoRaw, error, loading] = useGuestAgentInfo({ vmi, delay });
+  const guestAgentInfo = new GuestAgentInfoWrapper(guestAgentInfoRaw);
+  const fsList = guestAgentInfo.getFSInfo().getDisks();
+
+  const data =
+    guestAgentInfoRaw &&
+    fsList &&
+    fsList.map((fs, uid) => ({
+      metadata: {
+        uid,
+        name: fs.getDiskName(),
+        fileSystemType: fs.getType(),
+        mountPoint: fs.getMountPoint(),
+        totalBytes: humanizeBinaryBytes(fs.getTotalBytes()).string,
+        usedBytes: humanizeBinaryBytes(fs.getUsedBytes()).string,
+      },
+    }));
+
+  const body = () => {
+    if (vmStatusBundle.status !== VMStatus.RUNNING) {
```

Comment:
can we replace this with the helper `getGuestAgentFieldNotAvailMsg` ?

Reply:
Done!